pkg/controllers/management/nodepool/nodepool.go

```diff
@@ -207,10 +212,25 @@ func (c *Controller) createOrCheckNodes(nodePool *v3.NodePool, simulate bool) (b
 		if v3.NodeConditionProvisioned.IsFalse(node) || v3.NodeConditionInitialized.IsFalse(node) || v3.NodeConditionConfigSaved.IsFalse(node) {
 			changed = true
 			if !simulate {
-				c.deleteNode(node, 2*time.Minute)
+				_ = c.deleteNode(node, 2*time.Minute)
+			}
+		}
+		// remove unreachable nodes
+		q := getTaint(node.Spec.InternalNodeSpec.Taints, &unReachableTaint)
+		if q != nil && !simulate && nodePool.Spec.SelfHeal {
+			if isNodeReadyUnknown(node) {
+
+				start := q.TimeAdded.Time
+				if time.Since(start) > time.Duration(nodeNotReadyTimeout) {
+					err = c.deleteNode(node, 0)
+					if err != nil {
+						return false, err
+					}
+				} else {
+					go c.requeue(time.Duration(nodeNotReadyTimeout), nodePool, node)
```

Comment:
Use a map to signify that a go routine is already running to watch this particular node